orchestra/views.py

```diff
@@ -183,6 +204,128 @@ def submit_task_assignment(request):
+                      in_progress_iterations=True)
+    project_information = get_project_information(project_id)
+    project_information['project']['admin_url'] = urljoin(
+        settings.ORCHESTRA_URL,
+        urlresolvers.reverse(
+            'admin:orchestra_project_change',
+            args=(project_id,)))
+
+    for slug, task in project_information['tasks'].items():
+        task['admin_url'] = urljoin(
+            settings.ORCHESTRA_URL,
+            urlresolvers.reverse(
+                'admin:orchestra_task_change',
+                args=(task['id'],)))
+
+        for assignment in task['assignments']:
+            assignment['admin_url'] = urljoin(
+                settings.ORCHESTRA_URL,
+                urlresolvers.reverse(
+                    'admin:orchestra_taskassignment_change',
+                    args=(assignment['id'],)))
+
+            iterations = df[(df.worker == assignment['worker']) &
+                            (df.task_id == task['id'])]
+            assignment['iterations'] = []
+            for idx, info in iterations.T.items():
+                iteration = info.to_dict()
+                iteration['end_datetime'] = (
+                    iteration['start_datetime'] + iteration['calendar_time'])
+                del iteration['calendar_time']
+                del iteration['work_time']
+                assignment['iterations'].append(iteration)
+    return project_information
+
+
+@json_view
+@login_required
+@user_passes_test(is_project_admin)
+def reassign_assignment_api(request):
+    worker_username = json.loads(request.body.decode())['worker_username']
+    worker = Worker.objects.get(user__username=worker_username)
+    assignment_id = json.loads(request.body.decode())['assignment_id']
+    reassign_assignment(worker.id, assignment_id)
+
+
+@json_view
+@login_required
+@user_passes_test(is_project_admin)
+def revert_task_api(request):
+    body = json.loads(request.body.decode())
+    # Python datetime expects seconds, while javascript provides milliseconds
+    revert_datetime = datetime.utcfromtimestamp(body['revert_datetime'] / 1000)
+    revert_datetime = pytz.utc.localize(revert_datetime)
+    audit = revert_task_to_datetime(body['taskId'],
+                                    revert_datetime,
+                                    body['fake'])
+
+    audit['task'] = TaskSerializer(audit['task']).data
+    for assignment in audit['assignments']:
+        assignment['assignment'] = TaskAssignmentSerializer(
+            assignment['assignment']).data
+
+    return audit
+
+
+@json_view
+@login_required
+@user_passes_test(is_project_admin)
+def complete_and_skip_task_api(request):
+    task_id = json.loads(request.body.decode())['task_id']
+    complete_and_skip_task(task_id)
+
+
+@json_view
+@login_required
+@user_passes_test(is_project_admin)
+def assign_task_api(request):
+    worker_username = json.loads(request.body.decode())['worker_username']
```

Comment:
Talked about offline; single object retrievals can live within the API methods as long as nothing else is modified.